core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -278,4 +278,10 @@ package object config {
         "spark.io.compression.codec.")
       .booleanConf
       .createWithDefault(false)
+  private[spark] val SHUFFLE_ACCURATE_BLOCK_THRESHOLD =
+    ConfigBuilder("spark.shuffle.accurateBlkThreshold")
+      .doc("This only works in HighlyCompressedMapStatus. Sizes of blocks above this threshold" +
+        " will be recorded accurately. This is to help control memory cost when shuffle blocks.")
```

Comment:
how about
```
When we compress the size of shuffle blocks in HighlyCompressedMapStatus, we will record the size accurately if it's above the threshold specified by this config. This helps to prevent OOM by avoiding underestimating shuffle block size when fetching shuffle blocks.
```